proto/substrait/algebra.proto

```diff
@@ -338,6 +339,13 @@ message SetRel {
     SET_OP_INTERSECTION_MULTISET = 4;
     SET_OP_UNION_DISTINCT = 5;
     SET_OP_UNION_ALL = 6;
```

Comment:
With this approach, could you added a deprecated flag to `SET_OP_UNION_DISTINCT` and `SET_OP_UNION_UNION_ALL`, along with a comment indicating that folks should use `SET_OP_UNION` in conjunction with the `mode` field instead.